enclave_manager/Dockerfile

```diff
@@ -137,61 +121,136 @@ RUN OPENSSL_VER=1.1.1d \
         echo "ftp_proxy=$ftp_proxy" >> /etc/environment; \
     fi \
  && if [ ! -z "$https_proxy" ]; then \
-        echo 'Acquire::https::Proxy "'$https_proxy'";' \
-              >> /etc/apt/apt.conf.d/00proxy; \
+    echo 'Acquire::https::Proxy "'$https_proxy'";' >> /etc/apt/apt.conf.d/00proxy; \
         echo "https_proxy=$https_proxy" >> /etc/wgetrc; \
         echo "https_proxy=$https_proxy" >> /etc/environment; \
-        echo "HTTPS_PROXY=$(echo $https_proxy | sed 's,[a-zA-Z]*://,,')" \
-             >> /etc/environment; \
+        echo "HTTPS_PROXY=$(echo $https_proxy | sed 's,[a-zA-Z]*://,,')" >> /etc/environment; \
     fi
 
 ENV http_proxy=$http_proxy
 ENV https_proxy=$https_proxy
-ENV TCF_HOME=/project/TrustedComputeFramework
+ENV TCF_HOME=/project/avalon
 ENV SGX_SSL=/opt/intel/sgxssl
 ENV SGX_SDK=/opt/intel/sgxsdk
 ENV PATH=$PATH:$SGX_SDK/bin:$SGX_SDK/bin/x64
 ENV PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$SGX_SDK/pkgconfig
 ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$SGX_SDK/sdk_libs
-# Set display, replace value with [IP address of host]:0
-ENV DISPLAY=:0
+ENV TCF_ENCLAVE_CODE_SIGN_PEM="$TCF_HOME/enclave.pem"
+
+RUN openssl genrsa -3 -out $TCF_HOME/enclave.pem 3072
+
+WORKDIR /project/avalon/common/sgx_workload
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
+
+WORKDIR /project/avalon/examples/apps
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
+
+WORKDIR /project/avalon/common/cpp
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
+
+WORKDIR /project/avalon/tc/sgx/trusted_worker_manager/common
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
+
+WORKDIR /project/avalon/tc/sgx/trusted_worker_manager/enclave
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
 
-WORKDIR /project/TrustedComputeFramework/tools/build
+WORKDIR /project/avalon/tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge
 
-RUN make lint && echo "LINT SUCCESS"
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
 
-RUN if [ "$MAKECLEAN" != 0 ]; then echo "Clean build" && make clean; fi \
- && make \
- && echo "BUILD SUCCESS"
+WORKDIR /project/avalon/common/python
+
+RUN make \
+  && make install
+
+#ENV PATH=$PATH:/project/avalon/examples/enclave_manager
+
+WORKDIR /project/avalon/sdk
+
+RUN python3 setup.py bdist_wheel
+
+WORKDIR /project/avalon/enclave_manager
+
+RUN echo "Building avalon enclave manager" \
+  && make
+#  && make install
+
+# Build Final image and install dependent modules
+FROM base_image as final_image
+
+ENV http_proxy=$http_proxy
+ENV https_proxy=$https_proxy
+ENV TCF_HOME=/project/avalon
+
+WORKDIR /project/avalon/
+
+COPY --from=build_image /usr/local/ /usr/local
+COPY --from=build_image /project/avalon/config/tcs_config.toml /project/avalon/config/
+COPY --from=build_image /project/avalon/tc/sgx/trusted_worker_manager/enclave/build/ /project/avalon/tc/sgx/trusted_worker_manager/enclave/build
```

Comment:
We don;t need to copy entire build artifacts. We need to copy only required files like shared library and executable ?